Ron

RonTang opened the pull request:
performTaskInCocosThread and performFunctionInCocosThread difference is that we can use performTaskInCocosThread return-value to determine , whether submitted to CocosThread function is finished, and get submitted to CocosThread function's return-value .
When we need to perform some operations in other threads , these thread need wait for cocos thread update over, this method provides convenience.
why our function return 'bool',because of Microsoft bug https://connect.microsoft.com/VisualStudio/feedback/details/791185/std-packaged-task-t-where-t-is-void-or-a-reference-class-are-not-movable,at current,we must need function returning a value like 'bool'.
It is ok in ubuntu 12.04 and window 7,other platform not test.

performTaskInCocosThread与performFunctionInCocosThread区别在于，我们可以通过performTaskInCocosThread返回值确定，提交给CocosThread的函数是否执行完毕，并获取提交给CocosThread函数的返回值。为什么提交函数要返回bool，因为Microsoft的bug，https://connect.microsoft.com/VisualStudio/feedback/details/791185/std-packaged-task-t-where-t-is-void-or-a-reference-class-are-not-movable。暂时我们需要返回值
当我们需要在其他线程执行一些操作，并且要等待cocosThread更新完成时，此方法提供便利。在ubuntu和window下测试正常，其他平台未测试。

Commit:
cocos2dx: CCScheduler

Add a function name as "performTaskInCocosThread".
calls a function on the cocos2d thread.
Useful when you need to call a cocos2d function from another thread.
Note that return value indicate function running state in cocos2d thread.
This function is thread safe.

Signed-off-by: tangzhenqiang <tangzhenqiang@letv.com>

Commit:
base: CCScheduler.h

Completion Notes of CCScheduler::performTaskInCocosThread.

Signed-off-by: tangzhenqiang <tangzhenqiang@letv.com>

RonTang commented:
this function  are not compatible with lua.

RonTang commented:
和自动生成的lua方法不兼容

samuele3hu commented:
@RonTang,you should skip binding this function by modify tools/tolua/cocos2dx.ini, as follows:

Add `performTaskInCocosThread` at the end of `
Scheduler::[pause resume unschedule schedule update isTargetPaused isScheduled performFunctionInCocosThread]`

RonTang commented:
Thank you very much！I will try again.This is my first time to contribute my code. Thank you for your guidance！~谢谢@samuele3hu

Commit:
tolua: cocos2dx.ini

modified cocos2dx.ini to skip binding Scheduler performTaskInCocosThread
with lua.

Signed-off-by: tangzhenqiang <tangzhenqiang@letv.com>

minggo commented:
i am not sure if we need this new function, but i don't find any other developer ask for this feature, so close this PR
